lib/ddtrace/method_tracing.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Datadog
+  # \MethodTracing adds simple helpers for adding
+  # tracing to your methods.
+  #
+  # = Example
+  #
+  #   extend Datadog::MethodTracing
+  #   trace_methods self, :method1, :method2
+  module MethodTracing
+    # Wrap *methods* on *klass* in a trace block
+    def trace_methods(klass, *methods)
+      methods.each { |m| trace_method(klass, m) }
+    end
+
+    def trace_method(klass, method)
+      metric = "#{class_name(klass)}.#{method}"
+      m = Module.new do
+        define_method(method) do |*args, &block|
+          Datadog.tracer.trace(metric) { super(*args, &block) }
```

Comment:
I believe you can simplify parameter delegation here:
```suggestion
          Datadog.tracer.trace(metric) { super }
```

Reply:
No, you cannot omit the `super` arguments inside `define_method`:
```
ruby -e "define_method(:foo) { super } ; foo"
-e:1:in `block in <main>': implicit argument passing of super from method defined by define_method() is not supported. Specify all arguments explicitly. (RuntimeError)
	from -e:1:in `<main>'
```

Reply:
Thank you @bquorning! Removing this suggestion.